### PR TITLE
fix issue with dependent package purl on deps.dev and remove duplicates for vulnerability and hasSourceAt

### PR DESCRIPTION
**Pull request by pxp928**
Signed-off-by: pxp928 <parth.psu@gmail.com>

- Also adds edge filters to certifier queries
- remove vulnerability collection from deps.dev as it ends up with duplicate data from the osv certifier
- check if a source has already been ingested for a purl. If it has, do not re-ingest as it duplicates "hasSourceAt" (different timestamps)

### CODE REVIEW

File: internal/testing/testdata/testdata.go [hunk 1]
```diff
@@ -982,9 +982,7 @@ var (
 			  "CurrentPackage":{
 				 "name":"foreign-types-shared",
 				 "namespace":"",
-				 "qualifiers":[
-					
-				 ],
+				 "qualifiers":null,
```
Comment (lumjjb) [inline internal/testing/testdata/testdata.go:985]: do not merge until we resolve issue with duplicates from deps.dev

Comment (pxp928) [reply internal/testing/testdata/testdata.go:984]: Ready to review! No more duplicates for `hasSourceAt` and `CertifyVuln`

**Review by mihaimaruseac — APPROVED** [review]

File: pkg/ingestor/parser/deps_dev/deps_dev.go [hunk 2]
```diff
@@ -182,5 +138,16 @@ func (d *depsDevParser) GetIdentities(ctx context.Context) []common.TrustInforma
 }
 
 func (d *depsDevParser) GetIdentifiers(ctx context.Context) (*common.IdentifierStrings, error) {
-	return nil, fmt.Errorf("not yet implemented")
+	idstrings := &common.IdentifierStrings{}
+	for _, depComp := range d.packComponent.DepPackages {
```
Comment (pxp928) [inline pkg/ingestor/parser/deps_dev/deps_dev.go:142]: This does cause deps.dev to spin for a long time. Do we want this for the beta release?

Comment (pxp928) [reply pkg/ingestor/parser/deps_dev/deps_dev.go:142]: Future implementation would be that people would connect to the public instance to get this data and not keep going down the chain?

**Review by lumjjb — APPROVED** [review]

File: pkg/ingestor/parser/deps_dev/deps_dev.go (hunk 2 above)
Comment (lumjjb) [reply pkg/ingestor/parser/deps_dev/deps_dev.go:142]: deps.dev can handle pretty big QPS from my knowledge, however, i think based off https://github.com/google/deps.dev/issues/12, it seems like we have all the information we need to construct all that is needed... so if we implement a cache check in the deps.dev collector it should not result in a big hit rate (like for the sourceinput one but we can make it more generic if we have all the data available).

Comment (pxp928) [reply pkg/ingestor/parser/deps_dev/deps_dev.go]: Based on discussion, commented out the csub identifiers from deps.dev parser until the issue https://github.com/guacsec/guac/issues/769 is addressed.

**Pull request merged**